### PR TITLE
[Yoga] Add Octavia volume-based Amphora test

### PR DESCRIPTION
**Pull request by dosaboy**
Enable Amphora to run instances booted from
volume by flipping the config option
`enable-volume-based-amphora` to True and
checking volumes attached to the amphora
instances.

Depends-On: https://review.opendev.org/c/openstack/charm-octavia/+/931587
Related-Bug: https://launchpad.net/bugs/1901732

### CODE REVIEW

**Review by nicolasbock — APPROVED** [review]

**Review by nicolasbock — APPROVED** [review]

File: unit_tests/utilities/test_zaza_utilities_os_versions.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class TestOpenStackUtils(ut_utils.BaseTestCase):
+
+    def test_compare_openstack(self):
+        yoga = os_versions.CompareOpenStack('yoga')
+        xena = os_versions.CompareOpenStack('xena')
+        wallaby = os_versions.CompareOpenStack('wallaby')
+        self.assertGreater(yoga, xena)
+        self.assertLess(xena, yoga)
+        self.assertGreaterEqual(xena, xena)
+        self.assertGreaterEqual(yoga, yoga)
```
Comment (brianphaley) [inline unit_tests/utilities/test_zaza_utilities_os_versions.py:29]: Can the above ever fail? Seems like it was supposed to be different versions?

File: unit_tests/utilities/test_zaza_utilities_os_versions.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class TestOpenStackUtils(ut_utils.BaseTestCase):
+
+    def test_compare_openstack(self):
+        yoga = os_versions.CompareOpenStack('yoga')
+        xena = os_versions.CompareOpenStack('xena')
+        wallaby = os_versions.CompareOpenStack('wallaby')
+        self.assertGreater(yoga, xena)
+        self.assertLess(xena, yoga)
```
Comment (brianphaley) [inline unit_tests/utilities/test_zaza_utilities_os_versions.py:27]: Can these ever be equal? Based on L30 it seems they can?

**Review by brianphaley — COMMENTED** [review]
If this is a cherry-pick you can probably ignore my comments

File: unit_tests/utilities/test_zaza_utilities_os_versions.py (hunk 1 above)
Comment (brianphaley) [reply unit_tests/utilities/test_zaza_utilities_os_versions.py:29]: Talked to Ed, this was just updated based on where it was cherry-picked from.

**Pull request merged**